cpp/src/arrow/util/thread_pool.cc

```diff
@@ -58,6 +60,19 @@ SerialExecutor::~SerialExecutor() = default;
 
 Status SerialExecutor::SpawnReal(TaskHints hints, FnOnce<void()> task,
                                  StopToken stop_token, StopCallback&& stop_callback) {
+#ifdef ARROW_WITH_OPENTELEMETRY
+  // Wrap the task to propagate a parent tracing span to it
+  struct {
+    void operator()() {
+      auto scope = ::arrow::internal::tracing::GetTracer()->WithActiveSpan(activeSpan);
+      std::move(func)();
+    }
+    FnOnce<void()> func;
+    opentelemetry::nostd::shared_ptr<opentelemetry::trace::Span> activeSpan;
+  } wrapper{std::forward<FnOnce<void()>>(task),
```

Comment:
is `std::forward` the right call here, or just `std::move`?

Reply:
I changed this, and the equivalent in `future.cc`, into `std::move` (which is also used in another task wrapper I saw in `future.cc`)